### PR TITLE
feat: add validation to editableField

### LINKED ISSUE #1367: Form Validation - Github URL
### Overview
Currently, the projects details form doesn't have any validation. It increases onboarding time because user needs to go back and frequently correct typos (Form validation)

The project’s form fields need data validation, as opposed to them being free text fields.
Updated form should look as follows:    

- [ ] Github URL: add in validation -- https://www.github.com/[example or project name] - this will be a placeholder that will allow the user to copy and paste the final URL.  Placeholder will be red if it is incorrect. Developers to find the best way to validate. 

### Discussion 
- Error message: Generic error message "Doesn't fit criteria"

### Resources/Instructions
Usability Zoom Recording Timestamp 3:35
https://drive.google.com/drive/u/0/folders/1oSUAaU-fzKHpXeZpou5PH3SwC5eRnEWq

### PR DESCRIPTION
**Pull request by evanyang1**
Fixes #1367 #1368 #1369 #1370 #1363 #1402 

### What changes did you make and why did you make them ?

  - Added validation to editableField. Throws alert if the changes are rejected.
  - Added validation to description.
  - Upon onBlur(), the field value is trimmed and updated as trimmed.

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
image

</details>

### CODE REVIEW

**Comment by trillium** [comment]
Hey Evan,

Can you add a bit of context on the changes you've made to the codebase and why they've been made?

I know that we have the issue markes, and that you've referenced them within this PR, but it's good practice to put a summary together so that whoever is reviewing it has an easier time following along with the changes you've submitted.

File: client/src/components/manageProjects/editableField.js
Comment (trillium) [inline client/src/components/manageProjects/editableField.js:1]: We're planning on dropping the edits to editableField.js and keeping the utils validator

**Comment by trillium** [comment]
@evanyang1 
We're planning on merging #1401 first which will generate merge conflits. Pause developing on this until that is merged so we can handle easier.

**Comment by StephenTheDev1001** [comment]
You might want to add #1364  to the issues it fixes

**Review by trillium — APPROVED** [review]
Thanks for the validation functions Evan!

**Pull request merged**